solutions/secure/variables.tf

```diff
@@ -0,0 +1,156 @@
+  default     = []
+}
+
+variable "members" {
+  type        = number
+  description = "Allocated number of members. For more information, see https://cloud.ibm.com/docs/databases-for-elasticsearch?topic=databases-for-elasticsearch-resources-scaling"
+  default     = 3
+}
+
+variable "member_memory_mb" {
+  type        = number
+  description = "Allocated memory per-member. For more information, see https://cloud.ibm.com/docs/databases-for-elasticsearch?topic=databases-for-elasticsearch-resources-scaling"
+  default     = 1024
+}
+
+variable "member_cpu_count" {
+  type        = number
+  description = "Allocated dedicated CPU per member. For shared CPU, set to 0. For more information, see https://cloud.ibm.com/docs/databases-for-elasticsearch?topic=databases-for-elasticsearch-resources-scaling"
+  default     = 0
+}
+
+variable "member_disk_mb" {
+  type        = number
+  description = "Allocated disk per-member. For more information, see https://cloud.ibm.com/docs/databases-for-elasticsearch?topic=databases-for-elasticsearch-resources-scaling"
+  default     = 5120
+}
+
+variable "service_credential_names" {
+  description = "Map of name, role for service credentials that you want to create for the database"
+  type        = map(string)
+  default     = {}
+}
+
+variable "admin_pass" {
+  type        = string
+  description = "The password for the database administrator. If the admin password is null then the admin user ID cannot be accessed. More users can be specified in a user block."
+  default     = null
+  sensitive   = true
+}
+
+variable "users" {
+  type = list(object({
+    name     = string
+    password = string # pragma: allowlist secret
+    type     = string # "type" is required to generate the connection string for the outputs.
+    role     = optional(string)
+  }))
+  default     = []
+  sensitive   = true
+  description = "A list of users that you want to create on the database. Multiple blocks are allowed. The user password must be in the range of 10-32 characters. Be warned that in most case using IAM service credentials (via the var.service_credential_names) is sufficient to control access to the Elasticsearch instance. This blocks creates native Elasticsearch database users, more info on that can be found here https://cloud.ibm.com/docs/databases-for-elasticsearch?topic=databases-for-elasticsearch-user-management&interface=ui"
+}
+
+variable "tags" {
+  type        = list(any)
+  description = "Optional list of tags to be added to the Elasticsearch instance."
+  default     = []
+}
+
+variable "kms_key_crn" {
+  type        = string
+  description = "The root key CRN of the Hyper Protect Crypto Service (HPCS) to use for disk encryption."
+}
+
+variable "existing_kms_instance_guid" {
+  description = "The GUID of the Hyper Protect Crypto Services instance."
+  type        = string
+}
```

Comment:
This is not always required - its only required if creating the auth policy. Looks like its a bug n the fscloud submodule, so please fix there. It should have a default value of null, and the description should mention its only required if creating auth policy. You probably want to add some locals validation in the module to ensure it is not null if `skip_iam_authorization_policy` is false

Reply:
The root module seems to have that validation in the locals block